theforeman/foreman
# Fixes #36869 - Redirect /hosts to /new/hosts in index

Also removed logic in the hosts controller and application helper to rewrite the URL. Decided to go with the redirect on index approach instead since we want this redirect to happen only on get requests.

Test this
- Go to Administer-> Settings -> New Hosts Page. Set that to Yes
- Go to `http://<satellite>/hosts` and notice that it should get redirected to `http://<satellite>/new/hosts`
- Make operations like create/update/delete hosts all work ok.
